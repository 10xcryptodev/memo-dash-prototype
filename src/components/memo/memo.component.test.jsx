import React from 'react'
import { shallow } from 'enzyme'
import MemoComponent from './memo.component'

describe('<MemoComponent />', () => {
  let wrapper

  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  it('renders without crashing', () => {
    wrapper = shallow(<MemoComponent memo={'memo'} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render memo', () => {
    const memo = [
      {
        idx: 1,
        memoDatetime: '2018-07-02T14:18:42.728Z',
        memoLikesCount: 0,
        memoRepliesCount: 0,
        memoText: 'Iusto possimus quidem modi quis.',
        memoTipTotal: 0,
        username: 'alice'
      }
    ]

    wrapper = shallow(<MemoComponent memo={memo} />)
    expect(wrapper).toMatchSnapshot()
  })
})
