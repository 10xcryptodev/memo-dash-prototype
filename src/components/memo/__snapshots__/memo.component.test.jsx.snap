// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoComponent /> should render foreign memo 1`] = `
<Segment
  className="memo"
>
  <Comment>
    <CommentAvatar
      as="a"
      onClick={[Function]}
      src="https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg"
    />
    <CommentContent>
      <CommentAuthor
        as="a"
        onClick={[Function]}
      >
        alice
      </CommentAuthor>
      <CommentMetadata>
        <span>
          2018-07-26T16:44:56.000Z
        </span>
      </CommentMetadata>
      <CommentText>
        Suscipit vel nobis.
      </CommentText>
      <CommentActions>
        <CommentAction
          as="a"
          name="replyAction"
          onClick={[Function]}
        >
          <Icon
            as="i"
            name="reply"
          />
          <span>
            0
          </span>
        </CommentAction>
        <CommentAction
          as="a"
          name="likeAction"
          onClick={[Function]}
        >
          <Icon
            as="i"
            color="grey"
            name="like"
          />
          <span>
            1
          </span>
        </CommentAction>
        <CommentAction
          as="a"
          name="tipAction"
        >
          <Icon
            as="i"
            color="grey"
            name="btc"
          />
          <span>
            0
          </span>
        </CommentAction>
      </CommentActions>
    </CommentContent>
  </Comment>
</Segment>
`;

exports[`<MemoComponent /> should render memo with reply form 1`] = `
<Segment
  className="memo"
>
  <Comment>
    <CommentAvatar
      className="avatar"
      src="https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg"
    />
    <CommentContent>
      <CommentAuthor
        as="a"
        onClick={[Function]}
      >
        alice
      </CommentAuthor>
      <CommentMetadata>
        <span>
          2018-07-26T16:44:56.000Z
        </span>
      </CommentMetadata>
      <CommentText>
        Suscipit vel nobis.
      </CommentText>
      <CommentActions>
        <CommentAction
          as="span"
          name="replyAction"
        >
          <Icon
            as="i"
            name="reply"
          />
          <span>
            0
          </span>
        </CommentAction>
        <CommentAction
          as="span"
          name="likeAction"
        >
          <Icon
            as="i"
            color="grey"
            name="like"
          />
          <span>
            1
          </span>
        </CommentAction>
        <CommentAction
          as="a"
          name="tipAction"
        >
          <Icon
            as="i"
            color="grey"
            name="btc"
          />
          <span>
            0
          </span>
        </CommentAction>
      </CommentActions>
      <WithFormik(ReplyFormComponent)
        memoId={1}
        onReplyClicked={[Function]}
        username="alice"
      />
    </CommentContent>
  </Comment>
</Segment>
`;

exports[`<MemoComponent /> should render own memo 1`] = `
<Segment
  className="memo"
>
  <Comment>
    <CommentAvatar
      className="avatar"
      src="https://s3.amazonaws.com/uifaces/faces/twitter/chaabane_wail/128.jpg"
    />
    <CommentContent>
      <CommentAuthor
        as="a"
        onClick={[Function]}
      >
        alice
      </CommentAuthor>
      <CommentMetadata>
        <span>
          2018-07-26T16:44:56.000Z
        </span>
      </CommentMetadata>
      <CommentText>
        Suscipit vel nobis.
      </CommentText>
      <CommentActions>
        <CommentAction
          as="span"
          name="replyAction"
        >
          <Icon
            as="i"
            name="reply"
          />
          <span>
            0
          </span>
        </CommentAction>
        <CommentAction
          as="span"
          name="likeAction"
        >
          <Icon
            as="i"
            color="grey"
            name="like"
          />
          <span>
            1
          </span>
        </CommentAction>
        <CommentAction
          as="a"
          name="tipAction"
        >
          <Icon
            as="i"
            color="grey"
            name="btc"
          />
          <span>
            0
          </span>
        </CommentAction>
      </CommentActions>
    </CommentContent>
  </Comment>
</Segment>
`;

exports[`<MemoComponent /> should render without crashing 1`] = `
<Segment
  className="memo"
>
  <Comment>
    <CommentAvatar
      className="avatar"
    />
    <CommentContent>
      <CommentAuthor
        as="a"
        onClick={[Function]}
      />
      <CommentMetadata>
        <span />
      </CommentMetadata>
      <CommentText />
      <CommentActions>
        <CommentAction
          as="span"
          name="replyAction"
        >
          <Icon
            as="i"
            name="reply"
          />
          <span />
        </CommentAction>
        <CommentAction
          as="span"
          name="likeAction"
        >
          <Icon
            as="i"
            color="grey"
            name="like"
          />
          <span />
        </CommentAction>
        <CommentAction
          as="a"
          name="tipAction"
        >
          <Icon
            as="i"
            color="grey"
            name="btc"
          />
          <span />
        </CommentAction>
      </CommentActions>
    </CommentContent>
  </Comment>
</Segment>
`;
